In [1]:
import pandas as pd

df = pd.read_csv("D:/My Projects/experiment-uplifting engine/data/experiment_simulated.csv")
df.head()

ProductID ProductName     Category   Price  Rating  NumReviews  \
0          1  Headphones  Electronics  400.31     1.7        3772   
1          2  Headphones  Electronics  235.03     2.3        2919   
2          3  Smartwatch  Electronics  417.90     1.8        1184   
3          4  Smartphone  Electronics  152.70     3.4        2047   
4          5      Laptop  Electronics  394.74     1.8        1267   

   StockQuantity  Discount  Sales   DateAdded         City  prior_purchases  \
0             20      0.08    466  2023-11-10  Albuquerque                8   
1            663      0.33   1332  2023-09-28       Queens                6   
2            459      0.31    252  2023-10-20       Austin                2   
3            475      0.49   1806  2023-12-05       Dallas                4   
4            831      0.23   1508  2023-08-10       Tucson                3   

   new_user  high_value_user  treatment   device  treatment_effect  \
0         0                1          0   mobile              0.01   
1         0                1          1   mobile              0.01   
2         0                0          1  desktop              0.00   
3         0                0          1   mobile              0.02   
4         0                0          1   mobile              0.02   

   conversion_prob  converted    uplift  
0             0.08          0 -0.024481  
1             0.09          0  0.030882  
2             0.08          0 -0.002081  
3             0.10          0  0.023280  
4             0.10          0 -0.007054

In [2]:
df.groupby(['new_user','treatment'])['converted'].mean()


new_user  treatment
0         0            0.072368
          1            0.081466
1         0            0.035714
          1            0.160000
Name: converted, dtype: float64

In [3]:
df.groupby(['device','treatment'])['converted'].mean()


device   treatment
desktop  0            0.080000
         1            0.067961
mobile   0            0.063380
         1            0.096774
Name: converted, dtype: float64

In [5]:
avg_order_value = 100

if "df" not in globals():
    import pandas as pd
    df = pd.read_csv("D:/My Projects/experiment-uplifting engine/data/experiment_simulated.csv")

treatment_rate = df.loc[df["treatment"] == 1, "converted"].mean()
control_rate = df.loc[df["treatment"] == 0, "converted"].mean()
absolute_lift = treatment_rate - control_rate

incremental_revenue = absolute_lift * len(df) * avg_order_value
print("Absolute lift:", round(absolute_lift, 4))
print("Estimated incremental revenue:", round(incremental_revenue, 2))


Absolute lift: 0.015
Estimated incremental revenue: 1502.34


Based on statistically significant uplift of 0.015%
we recommend rolling out to new users and mobile users only.
